### DISCUSSION TITLE
install ERROR on M1 MacBook

### DISCUSSION THREAD

**Discussion by tshuangteng** [pymupdf/PyMuPDF#1704] [Looking for help]
% pip install PyMuPDF
Collecting PyMuPDF
  Downloading PyMuPDF-1.19.6.tar.gz (2.3 MB)
     |████████████████████████████████| 2.3 MB 1.2 MB/s            
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: PyMuPDF
  Building wheel for PyMuPDF (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /Users/ht/Venv/pdf2xls/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/4n/8f053jy17cbc8wnww48g8j0w0000gn/T/pip-install-fc2lo9iy/pymupdf_3e1a1995f57b4d5eabc9e3b6917d4fa2/setup.py'"'"'; __file__='"'"'/private/var/folders/4n/8f053jy17cbc8wnww48g8j0w0000gn/T/pip-install-fc2lo9iy/pymupdf_3e1a1995f57b4d5eabc9e3b6917d4fa2/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/4n/8f053jy17cbc8wnww48g8j0w0000gn/T/pip-wheel-09elteay
       cwd: /private/var/folders/4n/8f053jy17cbc8wnww48g8j0w0000gn/T/pip-install-fc2lo9iy/pymupdf_3e1a1995f57b4d5eabc9e3b6917d4fa2/
  Complete output (37 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-11-universal2-3.8
  creating build/lib.macosx-11-universal2-3.8/fitz
  copying fitz/__init__.py -> build/lib.macosx-11-universal2-3.8/fitz
  copying fitz/fitz.py -> build/lib.macosx-11-universal2-3.8/fitz
  copying fitz/utils.py -> build/lib.macosx-11-universal2-3.8/fitz
  copying fitz/__main__.py -> build/lib.macosx-11-universal2-3.8/fitz
  running build_ext
  creating build/temp.macosx-11-universal2-3.8
  creating build/temp.macosx-11-universal2-3.8/fitz
  gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch arm64 -arch x86_64 -g -I/usr/local/include/mupdf -I/usr/local/include -Imupdf/thirdparty/freetype/include -I/usr/local/include/freetype2 -I/usr/X11R6/include/freetype2 -I/opt/homebrew/include -I/opt/homebrew/include/mupdf -I/opt/homebrew/include/freetype2 -I/Users/ht/Venv/pdf2xls/include -I/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8 -c fitz/fitz_wrap.c -o build/temp.macosx-11-universal2-3.8/fitz/fitz_wrap.o
  fitz/fitz_wrap.c:1756:7: warning: 'tp_print' is deprecated [-Wdeprecated-declarations]
        0,                                    /* tp_print */
        ^
  /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/cpython/object.h:260:5: note: 'tp_print' has been explicitly marked deprecated here
      Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
      ^
  /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/pyport.h:515:54: note: expanded from macro 'Py_DEPRECATED'
  #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                       ^
  fitz/fitz_wrap.c:1923:7: warning: 'tp_print' is deprecated [-Wdeprecated-declarations]
        0,                                    /* tp_print */
        ^
  /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/cpython/object.h:260:5: note: 'tp_print' has been explicitly marked deprecated here
      Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
      ^
  /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/pyport.h:515:54: note: expanded from macro 'Py_DEPRECATED'
  #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                       ^
  fitz/fitz_wrap.c:2755:10: fatal error: 'fitz.h' file not found
  #include <fitz.h>
           ^~~~~~~~
  2 warnings and 1 error generated.
  error: command '/usr/bin/gcc' failed with exit code 1
  ----------------------------------------
  ERROR: Failed building wheel for PyMuPDF
  Running setup.py clean for PyMuPDF
Failed to build PyMuPDF
Installing collected packages: PyMuPDF
    Running setup.py install for PyMuPDF ... error
    ERROR: Command errored out with exit status 1:
     command: /Users/ht/Venv/pdf2xls/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/4n/8f053jy17cbc8wnww48g8j0w0000gn/T/pip-install-fc2lo9iy/pymupdf_3e1a1995f57b4d5eabc9e3b6917d4fa2/setup.py'"'"'; __file__='"'"'/private/var/folders/4n/8f053jy17cbc8wnww48g8j0w0000gn/T/pip-install-fc2lo9iy/pymupdf_3e1a1995f57b4d5eabc9e3b6917d4fa2/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/4n/8f053jy17cbc8wnww48g8j0w0000gn/T/pip-record-m_k3sx7p/install-record.txt --single-version-externally-managed --compile --install-headers /Users/ht/Venv/pdf2xls/include/site/python3.8/PyMuPDF
         cwd: /private/var/folders/4n/8f053jy17cbc8wnww48g8j0w0000gn/T/pip-install-fc2lo9iy/pymupdf_3e1a1995f57b4d5eabc9e3b6917d4fa2/
    Complete output (39 lines):
    running install
    /Users/ht/Venv/pdf2xls/lib/python3.8/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
      warnings.warn(
    running build
    running build_py
    creating build
    creating build/lib.macosx-11-universal2-3.8
    creating build/lib.macosx-11-universal2-3.8/fitz
    copying fitz/__init__.py -> build/lib.macosx-11-universal2-3.8/fitz
    copying fitz/fitz.py -> build/lib.macosx-11-universal2-3.8/fitz
    copying fitz/utils.py -> build/lib.macosx-11-universal2-3.8/fitz
    copying fitz/__main__.py -> build/lib.macosx-11-universal2-3.8/fitz
    running build_ext
    creating build/temp.macosx-11-universal2-3.8
    creating build/temp.macosx-11-universal2-3.8/fitz
    gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch arm64 -arch x86_64 -g -I/usr/local/include/mupdf -I/usr/local/include -Imupdf/thirdparty/freetype/include -I/usr/local/include/freetype2 -I/usr/X11R6/include/freetype2 -I/opt/homebrew/include -I/opt/homebrew/include/mupdf -I/opt/homebrew/include/freetype2 -I/Users/ht/Venv/pdf2xls/include -I/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8 -c fitz/fitz_wrap.c -o build/temp.macosx-11-universal2-3.8/fitz/fitz_wrap.o
    fitz/fitz_wrap.c:1756:7: warning: 'tp_print' is deprecated [-Wdeprecated-declarations]
          0,                                    /* tp_print */
          ^
    /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/cpython/object.h:260:5: note: 'tp_print' has been explicitly marked deprecated here
        Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
        ^
    /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/pyport.h:515:54: note: expanded from macro 'Py_DEPRECATED'
    #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                         ^
    fitz/fitz_wrap.c:1923:7: warning: 'tp_print' is deprecated [-Wdeprecated-declarations]
          0,                                    /* tp_print */
          ^
    /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/cpython/object.h:260:5: note: 'tp_print' has been explicitly marked deprecated here
        Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
        ^
    /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/pyport.h:515:54: note: expanded from macro 'Py_DEPRECATED'
    #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                         ^
    fitz/fitz_wrap.c:2755:10: fatal error: 'fitz.h' file not found
    #include <fitz.h>
             ^~~~~~~~
    2 warnings and 1 error generated.
    error: command '/usr/bin/gcc' failed with exit code 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Users/ht/Venv/pdf2xls/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/4n/8f053jy17cbc8wnww48g8j0w0000gn/T/pip-install-fc2lo9iy/pymupdf_3e1a1995f57b4d5eabc9e3b6917d4fa2/setup.py'"'"'; __file__='"'"'/private/var/folders/4n/8f053jy17cbc8wnww48g8j0w0000gn/T/pip-install-fc2lo9iy/pymupdf_3e1a1995f57b4d5eabc9e3b6917d4fa2/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/4n/8f053jy17cbc8wnww48g8j0w0000gn/T/pip-record-m_k3sx7p/install-record.txt --single-version-externally-managed --compile --install-headers /Users/ht/Venv/pdf2xls/include/site/python3.8/PyMuPDF Check the logs for full command output.

**Comment by better319** [comment]
use anaconda

**Reply by tshuangteng** [reply]
are you sure this will work?
```
PackagesNotFoundError: The following packages are not available from current channels:

  - pymupdf

Current channels:

  - https://repo.anaconda.com/pkgs/main/osx-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```
It seems to only work with x86 architecture computers...

**Reply by better319** [reply]
PyMuPDF ，not pymupdf

**Reply by tshuangteng** [reply]
额 你这素质也是可以... 你自己去试试再说,你以为我和你一样傻!

**Reply by tshuangteng** [reply]
```
(base) ht@tsMacBook ~ % conda install PyMuPDF
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - pymupdf

Current channels:

  - https://repo.anaconda.com/pkgs/main/osx-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

**Comment by better319** [comment]
https://github.com/pymupdf/PyMuPDF/discussions/875
